sunpy/map/mapbase.py

```diff
@@ -222,6 +223,9 @@ def __init__(self, data, header, plot_settings=None, **kwargs):
         if plot_settings:
             self.plot_settings.update(plot_settings)
 
+        if self._get_cmap_name() in sunpy_cm.cmlist:
```

Comment:
It's because `GenericMap.__init__()` is often called before `MyMapSource.__init__()` has fixed the wavelength metadata, or I suppose some maps might not have a wavelength at all